### DISCUSSION TITLE
Integrate React components as blocks directly.

### DISCUSSION THREAD

**Discussion by reyren18** [GrapesJS/grapesjs#5926] [Ideas]
Hey! I have hit a wall with this use case. There are certain plugins available but they fail to do the job when it comes to react components with external dependencies. A way to integrate react components directly as blocks would be very useful and much appreciated.

**Comment by Isc-mntl-snchz** [comment]
I'm not sure, but to do that you would need to load the libraries onto the canvas and have the component already configured. It would be quite interesting. I plan to do something similar with Vue.js, but I'm just starting to study how GrapesJS works. I have to admit that GrapesJS has exceeded my expectations. I will try to integrate Vue and will report the results.

**Comment by artf** [comment]
We have already an example of integrating React components https://codesandbox.io/p/sandbox/grapesjs-react-components-n6sff?file=%2Fsrc%2Findex.js (the CSS import preview of the CodeSandbox stopped working for some reason, but it works in standalone tab https://n6sff.csb.app)
